lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/IncrementalSummarizer.java

```diff
@@ -0,0 +1,320 @@
+    protected DoublePredicate predicate = d -> d != 0.0;
+
+    // Encoder
+    protected AttributeEncoder encoder = new AttributeEncoder();
+    protected List<Set<Integer>> inlierItemsets, outlierItemsets;
+
+    // Internal booking keep
+    private Deque<HashMap<Set<Integer>, Double>> inlierItemsetPaneCounts;
+    private Deque<HashMap<Set<Integer>, Double>> outlierItemsetPaneCounts;
+    private HashMap<Set<Integer>, Double> inlierItemsetPaneCount = new HashMap<>();
+    private HashMap<Set<Integer>, Double> outlierItemsetPaneCount = new HashMap<>();
+    private HashMap<Set<Integer>, Double> inlierItemsetWindowCount = new HashMap<>();
+    private HashMap<Set<Integer>, Double> outlierItemsetWindowCount = new HashMap<>();
+    private Deque<Integer> inlierPaneCounts;
+    private Deque<Integer> outlierPaneCounts;
+    private List<Integer> inlierCountCumSum;
+    private List<Integer> outlierCountCumSum;
+    private HashMap<Set<Integer>, Integer> trackingMap = new HashMap<>();
+
+    public IncrementalSummarizer(int numPanes) { setWindowSize(numPanes); }
+
+    @Override
+    public void setWindowSize(int numPanes) {
+        this.numPanes = numPanes;
+        if (inlierItemsetPaneCounts == null) {
+            inlierPaneCounts = new ArrayDeque<>(numPanes);
+            outlierPaneCounts = new ArrayDeque<>(numPanes);
+            inlierItemsetPaneCounts = new ArrayDeque<>(numPanes);
+            outlierItemsetPaneCounts = new ArrayDeque<>(numPanes);
+        }
+    }
+
+    @Override
+    public int getWindowSize() { return numPanes; }
+
+    public IncrementalSummarizer setMinSupport(double minSupport) {
+        this.minOutlierSupport = minSupport;
+        return this;
+    }
+
+    /**
+     * Adjust this to tune the severity (e.g. strength of correlation) of the results returned.
+     * @param minRiskRatio lowest risk ratio to consider for meaningful explanations.
+     * @return this
+     */
+    public IncrementalSummarizer setMinRiskRatio(double minRiskRatio) {
+        this.minRiskRatio = minRiskRatio;
+        return this;
+    }
+
+    /**
+     * By default, will check for nonzero entries in a column of doubles.
+     * @param predicate function to signify whether row should be treated as outlier.
+     * @return this
+     */
+    public IncrementalSummarizer setOutlierPredicate(DoublePredicate predicate) {
+        this.predicate = predicate;
+        return this;
+    }
+    public IncrementalSummarizer setAttributes(List<String> attributes) {
+        this.attributes = attributes;
+        this.encoder.setColumnNames(attributes);
+        return this;
+    }
+
+    /**
+     * Whether or not to use combinations of attributes in explanation, or only
+     * use simple single attribute explanations
+     * @param useAttributeCombinations flag
+     * @return this
+     */
+    public IncrementalSummarizer setUseAttributeCombinations(boolean useAttributeCombinations) {
```

Comment:
I just inherited this from the exisiting code base. I don't think we have a use case for this so happy to remove it.